### PR TITLE
don't panic on connection errors in DefaultRetryPolicy

### PR DESCRIPTION
**Pull request by alexeldeib**
added some unit tests as well.

### CODE REVIEW

File: api/client.go [hunk 1]
```diff
@@ -1060,5 +1060,8 @@ func DefaultRetryPolicy(ctx context.Context, resp *http.Response, err error) (bo
 	if err != nil || retry {
 		return retry, err
 	}
-	return resp.StatusCode == 412, nil
+	if resp != nil && resp.StatusCode == 412 {
```
Comment (alexeldeib) [inline api/client.go:1063]: I can inline the nil check, I found this more readable.

**Comment by ncabatoff** [comment]
Hi @alexeldeib,

Thanks for the bug report and PR.  Nice find and nice fix - I appreciate the new test, which I should've added myself.

**Review by ncabatoff — APPROVED** [review]

**Comment by ncabatoff** [comment]
I can't merge this PR because some tests are failing.  The fix is in the master branch (ba8ab5083fe2184e5c11455877e794d896f50b28), could you update your fork so the tests pass please?

**Comment by ncabatoff** [comment]
Thanks again!

**Pull request merged**